packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useEffect, useState } from 'react';
+import { Viewer, Cartesian2, EllipsoidGeodesic, EventHelper } from 'cesium';
+import { isNumber, get } from 'lodash';
+import { useMap } from '../map';
+
+import './scale-tracker.tool.css';
+
+export interface RScaleTrackerToolProps {
+  locale?: { [key: string]: string };
+}
+
+interface IScaleData {
+  barWidth?: number;
+  distanceLabel?: string;
+  lastLegendUpdate: number;
+}
+
+/* eslint-disable @typescript-eslint/no-magic-numbers */
+const distances = [
+  1,
+  2,
+  3,
+  5,
+  10,
+  20,
+  30,
+  50,
+  100,
+  200,
+  300,
+  500,
+  1000,
+  2000,
+  3000,
+  5000,
+  10000,
+  20000,
+  30000,
+  50000,
+  100000,
+  200000,
+  300000,
+  500000,
+  1000000,
+  2000000,
+  3000000,
+  5000000,
+  10000000,
+  20000000,
+  30000000,
+  50000000,
+];
+
+const updateDistanceLegendCesium = (
+  mapViewer: Viewer,
+  prevScaleData: IScaleData,
+  setScaleData: React.Dispatch<React.SetStateAction<IScaleData>>,
+  locale?: { [key: string]: string }
+): void => {
+  const metersUnit = get(locale, 'METERS_UNIT') ?? 'm';
+  const kiloMetersUnit = get(locale, 'KILOMETERS_UNIT') ?? 'km';
+  const scale: IScaleData = {
+    barWidth: undefined,
+    distanceLabel: undefined,
+    lastLegendUpdate: prevScaleData.lastLegendUpdate,
+  };
+  const geodesic = new EllipsoidGeodesic();
+
+  const now = new Date().getTime();
+  if (now < scale.lastLegendUpdate + 250) {
+    return;
+  }
+
+  scale.lastLegendUpdate = now;
+
+  // Find the distance between two pixels at the bottom center of the screen.
+  const width = mapViewer.scene.canvas.clientWidth;
+  const height = mapViewer.scene.canvas.clientHeight;
+
+  const left = mapViewer.scene.camera.getPickRay(
+    new Cartesian2((width / 2) | 0, height - 1)
+  );
+  const right = mapViewer.scene.camera.getPickRay(
+    new Cartesian2((1 + width / 2) | 0, height - 1)
+  );
+
+  const globe = mapViewer.scene.globe;
+  const leftPosition = globe.pick(left, mapViewer.scene);
+  const rightPosition = globe.pick(right, mapViewer.scene);
+
+  if (!leftPosition || !rightPosition) {
+    return;
+  }
+
+  const leftCartographic = globe.ellipsoid.cartesianToCartographic(
+    leftPosition
+  );
+  const rightCartographic = globe.ellipsoid.cartesianToCartographic(
+    rightPosition
+  );
+
+  geodesic.setEndPoints(leftCartographic, rightCartographic);
+  const pixelDistance = geodesic.surfaceDistance;
+
+  // Find the first distance that makes the scale bar less than 100 pixels.
+  const maxBarWidth = 100;
+  let distance;
+  for (let i = distances.length - 1; !isNumber(distance) && i >= 0; --i) {
+    if (distances[i] / pixelDistance < maxBarWidth) {
+      distance = distances[i];
+    }
+  }
```

Comment:
can we export the logic outside of the component? might be useful for other maps components

Reply:
Think about it, it's too much depends on CESIUM. Not sure it should be as a generic one.
Let's stick with local and due to the needs perform appropriate refactoring(hate this word)